GithubRepo/Data/Repository/GithubMainRepository.swift

```diff
@@ -6,47 +6,37 @@
 //
 
 import Foundation
+import RxRelay
+import RxSwift
 
 class GithubMainRepository: GithubRepository {
     private let dataSource: GithubDataSource
-    private(set) var repositories: [Repository]
-    private(set) var loading: Bool
-    private(set) var error: Bool
-    weak var delegate: GithubRepositoryDelegate?
+    private(set) var repositories: BehaviorRelay<[Repository]>
```

Comment:
it can be `let` instead of `private(set) var` as the subject will not change but its internal value will.

```suggestion
    let repositories: BehaviorRelay<[Repository]>
```